core/commands/add.go

```diff
@@ -49,7 +49,7 @@ remains to be implemented.
 	},
 
 	Arguments: []cmds.Argument{
-		cmds.FileArg("path", true, true, "The path to a file to be added to IPFS").EnableRecursive().EnableStdin(),
```

Comment:
On Mon, Apr 27, 2015 at 04:29:20PM -0700, Juan Batiz-Benet wrote:
> > -		cmds.FileArg("path", true, true, "The path to a file to be added to IPFS").EnableRecursive().EnableStdin(),
> 
> i think required still needs to be true. there's a complication here
> with the cmds lib because stdin counts as satisfying the
> requirement.

Hmm.  I think we need to make a distinction there, and maybe
automatically add the wrapping braces (‘[<path>]’) if EnableStdin() is
set (and auto-add a trailing ellipsis (‘[<path> …]’) if
EnableRecursive() is set.

> this comes from viewing stdin as just another file. which is what
> many cli tools do, but may not be the best model.

I like being able to add a file from stdin, but maybe don't bind the
two ideas (file paths from argument names or file content from stdin)
together so tightly.  Paths strings and an input stream aren't
interchangeable (e.g. we can read owner/group IDs from an input path,
but we lose that information if we pipe the file content into stdin).

So should I be PRing commands/argument.go instead?